### ISSUE TITLE
[RLLIB] Cloudpickle loading not working in windows but works in Linux

### ISSUE THREAD

**Issue by AvisP** [ray-project/ray#41002]
### What happened + What you expected to happen

I was trying to load a trained agent in rllib and getting an error `TypeError: code() argument 13 must be str, not int` on this [line](https://github.com/ray-project/ray/blob/50537bec9fd0bd00937e1d801e3cb4194e6d3bf4/rllib/connectors/agent/state_buffer.py#L41)

The full error traceback is

```
Exception has occurred: TypeError       (note: full exception trace is shown but execution is paused at: _run_module_as_main)
code() argument 13 must be str, not int
  File "C:\..........\Lib\site-packages\ray\rllib\connectors\agent\state_buffer.py", line 41, in __init__
    self._states = cloudpickle.loads(states)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\..........\Lib\site-packages\ray\rllib\connectors\agent\state_buffer.py", line 117, in from_state
    return StateBufferConnector(ctx, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\..........\Lib\site-packages\ray\rllib\connectors\registry.py", line 46, in get_connector
    return ALL_CONNECTORS[name].from_state(ctx, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\..........\Lib\site-packages\ray\rllib\connectors\agent\pipeline.py", line 68, in from_state
    raise e
  File "C:\..........\Lib\site-packages\ray\rllib\connectors\agent\pipeline.py", line 68, in from_state
    raise e
  File "C:\..........\Lib\site-packages\ray\rllib\connectors\registry.py", line 46, in get_connector
    return ALL_CONNECTORS[name].from_state(ctx, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\..........\Lib\site-packages\ray\rllib\connectors\util.py", line 135, in restore_connectors_for_policy
    return get_connector(name, ctx, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\..........\Lib\site-packages\ray\rllib\policy\policy.py", line 1037, in restore_connectors
    self.agent_connectors = restore_connectors_for_policy(
                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\..........\Lib\site-packages\ray\rllib\policy\policy.py", line 1085, in set_state
    self.restore_connectors(state)
  File "C:\..........\Lib\site-packages\ray\rllib\policy\torch_policy_v2.py", line 1091, in set_state
    super().set_state(state)
  File "C:\..........\Lib\site-packages\ray\rllib\policy\torch_mixins.py", line 114, in set_state
    super().set_state(state)
  File "C:\..........\Lib\site-packages\ray\rllib\evaluation\rollout_worker.py", line 1454, in set_state
    self.policy_map[pid].set_state(policy_state)
  File "C:\..........\Lib\site-packages\ray\rllib\algorithms\algorithm.py", line 2599, in __setstate__
    self.workers.local_worker().set_state(state["worker"])
  File "C:\..........\Lib\site-packages\ray\rllib\algorithms\algorithm.py", line 2157, in load_checkpoint
    self.__setstate__(checkpoint_data)
  File "C:\..........\Lib\site-packages\ray\tune\trainable\trainable.py", line 893, in restore
    self.load_checkpoint(to_load)
  File "C:\.......\RLLIB_google_cloud_train\rllibagent.py", line 213, in load
    self.nn_model.restore(checkpoint_path=os.path.join(last_chkt_path, last_chkt))
  File "C:\..........\Lib\site-packages\l2rpn_baselines\utils\gymAgent.py", line 112, in __init__
    self.load()
  File "C:\.......\RLLIB_google_cloud_train\rllibagent.py", line 164, in __init__
    super().__init__(g2op_action_space, gym_act_space, gym_obs_space,
  File "C:\......\RLLIB_google_cloud_train\evaluate_36bus.py", line 145, in evaluate
    grid2op_agent = RLLIBAgent(env.action_space,
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\......\RLLIB_google_cloud_train\evaluate_36bus.py", line 211, in <module>
    eval_agent, res_eval = evaluate(env,
                           ^^^^^^^^^^^^^
  File "C:\.......\AppData\Local\Programs\Python\Python311\Lib\runpy.py", line 88, in _run_code
    exec(code, run_globals)
  File "C:\........\AppData\Local\Programs\Python\Python311\Lib\runpy.py", line 198, in _run_module_as_main (Current frame)
    return _run_code(code, main_globals, None,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: code() argument 13 must be str, not int
```

### Versions / Dependencies

`rllib`:2.6.3
`cloudpickle`:2.2.1
`os`: Windows

### Reproduction script

I have attached the `states` file [here](https://file.io/WsZiaR3yDnIl)
```
import pickle
import cloudpickle

with open('states.pkl', 'rb') as F:
     loaded_object = pickle.load(F)
     
states = cloudpickle.load(loaded_objects)
```

Issue happening in Windows but working properly on Linux

### Issue Severity

High: It blocks me from completing my task.

**Comment by mattip** [comment]
Are you using the same version of python everywhere? What version of python did you use to create the pickle?
I found this [CPython issue](https://github.com/python/cpython/issues/100316) about loading code objects which led to my question about python versions
-

**Comment by AvisP** [comment]
So the original model was saved in linux running python `3.9.x`. I am having the issue on `3.11.6` on windows. I tried on python `3.10.11` on windows and getting the following error message when executing this `states = cloudpickle.load(loaded_object)`

```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: file must have 'read' and 'readline' attributes
``

**Comment by mattip** [comment]
So if I understand correctly, you are saving a cloudpickle on linux-python3.9, and it is failing to load into windows-python3.11 or windows-python3.10. I think the problem is that the problem is around including code objects, so a reproducer that includes the code to create the cloudpickle object is required.

**Comment by mattip** [comment]
> I was trying to load a trained agent in rllib ...

I am not sure loading a trained agent on a different system than the one used to save the agent is meant to work. @anyscalesam is that a scenario that rllib supports?

**Comment by AvisP** [comment]
Well the problem occurred when I used ray-lib to train a model on Google Cloud that runs linux-python3.9 and then I transferred the trained model to my windows machine to do inference. I debugged the issue and I found that is related to the the place where `cloudpickle` loading is happening. To keep things simple I described the problem in the above format.

This seems to be a common problem that maybe faced by lot of other users of ray-lib. Fortunately as a workaround with WSL I am able to infer it presently. I saw that the original data file got removed, so I will provide it [here](https://filebin.net/s21jsls4rgb8eaql) if you would like to investigate.
